lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js

```diff
@@ -322,16 +322,18 @@ module.exports = {
       'X-Amz-User-Agent',
     ];
 
-    let cors = {
-      origins: ['*'],
+    const cors = {
       origin: '*',
       methods: ['OPTIONS'],
       headers,
       allowCredentials: false,
     };
 
     if (typeof http.cors === 'object') {
-      cors = http.cors;
+      if (http.cors.origins) {
+        delete cors.origin;
+      }
+      Object.assign(cors, http.cors);
```

Comment:
Could you explain a little bit more your motivation behind assigning that? Why is it needed to assign the whole `http.cors` here?

Reply:
Even before this change, we have overwritten `cors` by `http.cors`. Now by assigning whole `http.cors` to `cors`, we can make sure that:
- Untouched following logics after this line will work in purpose.
- `cors` will still have default values in the case no corresponding properties are set in `http.cors`

@pgrzesik Could you guide me if I missed something here? :)

Reply:
Oh right, I missed that we were overriding `cors` object entirely in previous version, sorry for the confusion

Reply:
No problem at all! Thanks for quick reply :)